HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,343 @@
+        std::cout << "Could not open file for writing" << std::endl;
+    }
+
+    // Roll, Pitch and Yaw to Rotation Matrix
+    for(int i = 0; i < nofRotConv; i++)
+    {
+        std::cout << "\nConverting roll-pitch-yaw angles (" << representations.rotations[i].shortDesc
+                  << ") to rotation matrix" << std::endl;
+        Eigen::Matrix3d rotationMatrixFromRollPitchYaw =
+            rpy2rotmat(representations.rotations[i].rollPitchYaw, representations.rotations[i].convention);
+        std::cout << "Rotation Matrix:\n" << rotationMatrixFromRollPitchYaw << std::endl;
+
+        // Save Rotation Matrix to .yaml via OpenCV
+        fileStorageOut.write("RotationMatrixFrom" + representations.rotations[i].shortDesc,
+                             eigenToCv(rotationMatrixFromRollPitchYaw));
+
+        // Calculate error in transformation (Rotation Matrix --> Roll-Pitch-Yaw --> Rotation Matrix)
+        std::cout << "\nVerifying Rotation Matrix from Roll-Pitch-Yaw angles ("
+                  << representations.rotations[i].shortDesc << ")" << std::endl;
+        compareRotation(rotationMatrix, rotationMatrixFromRollPitchYaw);
+        compareRotationExtended(representations.rotations[i].rollPitchYaw,
+                                rotationMatrixFromRollPitchYaw,
+                                representations.rotations[i].convention);
+    }
+
+    // Rotation Vector to Axis-angle
+    std::cout << "\nConverting from rotation vector to axis-angle" << std::endl;
+    const Eigen::AngleAxisd axisAngle =
+        Eigen::AngleAxisd(representations.rotationVector.norm(), representations.rotationVector.normalized());
+    std::cout << "\nAxis-angle:\n" << reorderAxisAngle(axisAngle) << std::endl;
+
+    // Axis-Angle to Quaternion
+    std::cout << "\nConverting axis-angle to quaternion" << std::endl;
+    const Eigen::Quaterniond quaternion = Eigen::Quaterniond(axisAngle);
+    std::cout << "\nQuaternion:\n" << quaternion.coeffs() << std::endl;
+
+    // Quaternion to Rotation Matrix
+    std::cout << "\nConverting quaternion to rotation matrix" << std::endl;
+    const Eigen::Matrix3d rotationMatrixFromQuaternion = quaternion.toRotationMatrix();
+    std::cout << "\nRotation Matrix:\n" << rotationMatrixFromQuaternion << std::endl;
+    fileStorageOut.write("RotationMatrixFromQuaternion", eigenToCv(rotationMatrixFromQuaternion));
+
+    // Calculate error in transformation (Rotation Matrix --> Quaternion --> Rotation Matrix)
+    std::cout << "\nVerifying Rotation Matrix from Quaternion" << std::endl;
+    compareRotation(rotationMatrix, rotationMatrixFromQuaternion);
+
+    fileStorageOut.release();
+}
+
+Eigen::Vector4d reorderAxisAngle(const Eigen::AngleAxisd &axisAngle)
+{
+    Eigen::Vector4d reorderedAxisAngle;
+
+    for(int i = 0; i < 3; i++)
+        reorderedAxisAngle[i] = axisAngle.axis()[i];
+    reorderedAxisAngle[3] = axisAngle.angle();
+
+    return reorderedAxisAngle;
+}
+
+/** Converts Rotation Matrix to RollPitchYaw.
```

Comment:
`RollPitchYaw` is a specific type in your program, which this function doesn't return. Maybe you can write this comment a bit differently to avoid the confusion.